pkg/shell/nav.jsx

```diff
@@ -0,0 +1,201 @@
+import cockpit from "cockpit";
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+    Nav,
+    NavGroup,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon, ExclamationTriangleIcon, InfoCircleIcon } from '@patternfly/react-icons';
+import { OverlayTrigger, Tooltip } from 'patternfly-react';
+
+const _ = cockpit.gettext;
+
+export class CockpitNav extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            search: "",
+            current: props.current,
+        };
+
+        this.onSearchChanged = this.onSearchChanged.bind(this);
+        this.clearSearch = this.clearSearch.bind(this);
+    }
+
+    componentDidMount() {
+        // TODO - write this with react and not HTML API
```

Comment:
Is that something for this PR, or a follow-up?

Reply:
Follow up. It is rather complicated and I think it is just fine for now.